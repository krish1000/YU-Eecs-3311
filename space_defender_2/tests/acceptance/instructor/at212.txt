--Test Starfighter moving into projectile and enemy.
toggle_debug_mode
play(5,11,38,38,38,38,38)

setup_select(3)
setup_next(1)
setup_select(4)
setup_next(2)
setup_select(2)
setup_next(5)

move(A,1)
pass
special
move(A,10)

