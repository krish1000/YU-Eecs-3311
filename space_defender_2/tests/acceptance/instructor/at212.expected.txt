  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->toggle_debug_mode
  state:not started, debug, ok
  In debug mode.
->play(5,11,38,38,38,38,38)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_select(3)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Snipe
->setup_next(1)
  state:armour setup, debug, ok
  1:None
    Health:50, Energy:0, Regen:1/0, Armour:0, Vision:0, Move:1, Move Cost:0
  2:Light
    Health:75, Energy:0, Regen:2/0, Armour:3, Vision:0, Move:0, Move Cost:1
  3:Medium
    Health:100, Energy:0, Regen:3/0, Armour:5, Vision:0, Move:0, Move Cost:3
  4:Heavy
    Health:200, Energy:0, Regen:4/0, Armour:10, Vision:0, Move:-1, Move Cost:5
  Armour Selected:None
->setup_select(4)
  state:armour setup, debug, ok
  1:None
    Health:50, Energy:0, Regen:1/0, Armour:0, Vision:0, Move:1, Move Cost:0
  2:Light
    Health:75, Energy:0, Regen:2/0, Armour:3, Vision:0, Move:0, Move Cost:1
  3:Medium
    Health:100, Energy:0, Regen:3/0, Armour:5, Vision:0, Move:0, Move Cost:3
  4:Heavy
    Health:200, Energy:0, Regen:4/0, Armour:10, Vision:0, Move:-1, Move Cost:5
  Armour Selected:Heavy
->setup_next(2)
  state:power setup, debug, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_select(2)
  state:power setup, debug, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
->setup_next(5)
  state:in game(0.0), debug, ok
  Starfighter:
    [0,S]->health:210/210, energy:160/160, Regen:4/7, Armour:11, Vision:22, Move:10, Move Cost:7, location:[C,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11
    A _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _
->move(A,1)
  state:in game(1.0), debug, ok
  Starfighter:
    [0,S]->health:210/210, energy:146/160, Regen:4/7, Armour:11, Vision:22, Move:10, Move Cost:7, location:[A,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
      score:0
  Enemy:
    [1,G]->health:100/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,11]
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [C,1] -> [A,1]
  Enemy Action:
  Natural Enemy Spawn:
    A Grunt(id:1) spawns at location [A,11].
      1  2  3  4  5  6  7  8  9 10 11
    A S  _  _  _  _  _  _  _  _  _  G
    B _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _
->pass
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:210/210, energy:160/160, Regen:4/7, Armour:11, Vision:22, Move:10, Move Cost:7, location:[A,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
      score:0
  Enemy:
    [1,G]->health:110/110, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,9]
    [2,G]->health:100/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,11]
  Projectile:
    [-1,<]->damage:15, move:4, location:[A,8]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) passes at location [A,1], doubling regen rate.
  Enemy Action:
    A Grunt(id:1) gains 10 total health.
    A Grunt(id:1) moves: [A,11] -> [A,9]
      A enemy projectile(id:-1) spawns at location [A,8].
  Natural Enemy Spawn:
    A Grunt(id:2) spawns at location [A,11].
      1  2  3  4  5  6  7  8  9 10 11
    A S  _  _  _  _  _  _  <  G  _  G
    B _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _
->special
  state:in game(3.0), debug, ok
  Starfighter:
    [0,S]->health:260/210, energy:110/160, Regen:4/7, Armour:11, Vision:22, Move:10, Move Cost:7, location:[A,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
      score:0
  Enemy:
    [1,G]->health:130/130, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,7]
    [2,G]->health:120/120, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,9]
  Projectile:
    [-1,<]->damage:15, move:4, location:[A,4]
    [-2,<]->damage:15, move:4, location:[A,6]
    [-3,<]->damage:15, move:4, location:[A,8]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-1) moves: [A,8] -> [A,4]
  Starfighter Action:
    The Starfighter(id:0) uses special, gaining 50 health.
  Enemy Action:
    A Grunt(id:1) gains 20 total health.
    A Grunt(id:2) gains 20 total health.
    A Grunt(id:1) moves: [A,9] -> [A,7]
      A enemy projectile(id:-2) spawns at location [A,6].
    A Grunt(id:2) moves: [A,11] -> [A,9]
      A enemy projectile(id:-3) spawns at location [A,8].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11
    A S  _  _  <  _  <  G  <  G  _  _
    B _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _
->move(A,10)
  state:in game(4.0), debug, ok
  Starfighter:
    [0,S]->health:2/210, energy:54/160, Regen:4/7, Armour:11, Vision:22, Move:10, Move Cost:7, location:[A,10]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
      score:4
  Enemy:
    [3,G]->health:100/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:T, location:[B,11]
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-1) moves: [A,4] -> [A,1]
      The projectile collides with Starfighter(id:0) at location [A,1], dealing 4 damage.
    A enemy projectile(id:-2) moves: [A,6] -> [A,2]
    A enemy projectile(id:-3) moves: [A,8] -> [A,7]
      The projectile collides with Grunt(id:1) at location [A,7], healing 15 damage.
  Starfighter Action:
    The Starfighter(id:0) moves: [A,1] -> [A,10]
      The Starfighter collides with enemy projectile(id:-2) at location [A,2], taking 4 damage.
      The Starfighter collides with Grunt(id:1) at location [A,7], trading 130 damage.
      The Grunt at location [A,7] has been destroyed.
      The Starfighter collides with Grunt(id:2) at location [A,9], trading 120 damage.
      The Grunt at location [A,9] has been destroyed.
  Enemy Action:
  Natural Enemy Spawn:
    A Grunt(id:3) spawns at location [B,11].
      1  2  3  4  5  6  7  8  9 10 11
    A _  _  _  _  _  _  _  _  _  S  _
    B _  _  _  _  _  _  _  _  _  _  G
    C _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _
